src/block-data-providers/bitcoin-core/provider.ts

```diff
@@ -0,0 +1,278 @@
+            return;
+        }
+
+        let height = state.indexedBlockHeight + 1;
+
+        for (height; height <= tipHeight; height++) {
+            const transactions = await this.processBlock(height);
+            for (const transaction of transactions) {
+                const { txid, vin, vout, blockHeight, blockHash } = transaction;
+                await this.indexTransaction(
+                    txid,
+                    vin,
+                    vout,
+                    blockHeight,
+                    blockHash,
+                );
+            }
+            state.indexedBlockHeight = height;
+            await this.setState(state);
+        }
+        this.isSyncing = false;
+    }
+
+    async getTipHeight(): Promise<number> {
+        try {
+            const body = {
+                method: 'getblockcount',
+                params: [],
+                ...this.createPartialRPCBody(),
+            };
+            return await this.request(body);
+        } catch (error) {
+            this.logger.log(`Error fetching block count`);
+            throw error;
+        }
+    }
+
+    async getBlockHash(height: number): Promise<string> {
+        try {
+            const body = {
+                method: 'getblockhash',
+                params: [height],
+                ...this.createPartialRPCBody(),
+            };
+            return await this.request(body);
+        } catch (error) {
+            this.logger.log(`Error fetching  block hash of height : ${height}`);
+            throw error;
+        }
+    }
+
+    async getBlock(hash: string, verbosity: number): Promise<Block> {
+        try {
+            const body = {
+                method: 'getblock',
+                params: [hash, verbosity],
+                ...this.createPartialRPCBody(),
+            };
+            return await this.request(body);
+        } catch (error) {
+            this.logger.log(`Error fetching block with block hash : ${hash}`);
+            throw error;
+        }
+    }
+
+    async getRawTransaction(
+        txid: string,
+        isVerbose: boolean,
+    ): Promise<BlockTransaction> {
+        try {
+            const body = {
+                method: 'getrawtransaction',
+                params: [txid, isVerbose],
+                ...this.createPartialRPCBody(),
+            };
+            return await this.request(body);
+        } catch (error) {
+            this.logger.log(
+                `Error fetching transaction with transaction id : ${txid}`,
+            );
+            throw error;
+        }
+    }
+
+    public async processBlock(height: number): Promise<Transaction[]> {
+        const parsedTransactionList: Transaction[] = [];
+        const blockHash = await this.getBlockHash(height);
+        this.logger.log(
+            `Processing block at height ${height}, hash ${blockHash}`,
+        );
+        const block = await this.getBlock(blockHash, 2);
+        for (const txn of block.tx) {
```

Comment:
Why not just skip the first transactions? (first transaction is always coinbase transaction)
Then we can remove all the extra logic to handle coinbase transactions

Reply:
Sure it would be, but how certain are we that it would be returned in order. The RPC doc states 'returns an Object with information about block <hash> and information about each transaction'. I believe having a safety check does no harm.

Reply:
I think we already discussed this and agreed to remove the coinbase check and skip the first transaction